scripts/check-sample-lnd-conf.sh

```diff
@@ -0,0 +1,172 @@
+#!/bin/bash
+
+# This script performs different checks on the sample-lnd.conf file:
+# 1. Checks that all relevant options of lnd are included.
+# 2. Verifies that defaults are labeled if there are also further examples.
+# 3. Checks that all default values of lnd are mentioned correctly, including
+#    empty defaults and booleans which are set to false by default.
+
+set -e
+
+TAGS="$1"
+CONF_FILE=${2:-sample-lnd.conf}
+
+# We are reading the default values of lnd from lnd --help. To avoid formatting
+# issues the width of the terminal is set to 240. This needs a workaround for
+# CI where we don't have an interactive terminal.
+if [ -t 0 ]; then
+    stty cols 240
+    LND_HELP="$(go run -tags="$TAGS" \
+        github.com/lightningnetwork/lnd/cmd/lnd --help)"" --end"
+else
+    tmux new-session -d -s simulated-terminal -x 240 -y 9999
+    tmux send-keys -t simulated-terminal.0 "go run -tags=\"$TAGS\" \
+        github.com/lightningnetwork/lnd/cmd/lnd --help; tmux wait -S run" ENTER
+    tmux wait-for run
+    LND_HELP="$(tmux capture-pane -t simulated-terminal.0 -p)"" --end"
+    tmux kill-session -t simulated-terminal
+fi
+
+# LND_OPTIONS is a list of all options of lnd including the equal sign, which
+# is needed to distinguish between booleans and other variables.
+# It is created by reading the first two columns of lnd --help. 
+LND_OPTIONS="$(go run -tags="$TAGS" \
+    github.com/lightningnetwork/lnd/cmd/lnd --help | \
+    awk '{
+        option=""; 
+        if ($1 ~ /^--/){option=$1};
+        if ($2 ~ /^--/){option=$2}; 
+        if (match(option,  /--[^=]+[=]*/) && 
+                substr(option, length(option)) != "-")
+            {printf "%s ", substr(option, RSTART, RLENGTH)}
+        } 
+        END { printf "%s", "--end"}')"
+
+# OPTIONS_NO_CONF is a list of all options without any expected entries in 
+# sample-lnd.conf. There's no validation needed for these options. 
+OPTIONS_NO_CONF="help lnddir configfile version litecoin.signet \
+    litecoin.signetchallenge litecoin.signetseednode end"
+
+
+# OPTIONS_NO_LND_DEFAULT_VALUE_CHECK is a list of options with default values
+# set, but there aren't any returned defaults by lnd --help. Defaults have to be
+# included in sample-lnd.conf but no further checks are performed.
```

Comment:
@feelancer21 I assume we have to maintain the list of special cases to handle by this checker whenever we add a new option to `lnd.conf`.
Is there a checklist you can recommend with steps to follow if we add a new `lnd` flag? Maybe we can take it and include it in the developer contribution guide. What do you think @guggero.